### PR TITLE
pin torchmetrics version

### PR DESCRIPTION
**Pull request by nithinraok**
Signed-off-by: nithinraok <nithinrao.koluguri@gmail.com>

# What does this PR do ?

Fix torchmetrics version, as latest version has issues with pytorch Dataloader loading time. 

**Collection**: ASR

# Changelog 
- pin torchmetrics version to 0.10.3 
- revert accuracy API to 0.10.3 from 0.11.0 
- revert Speaker Jenkins run

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

### CODE REVIEW

**Comment by titu1994** [comment]
Why set upperbound instead of fixing it ? Future PTL versions may not be compatible with this torchmetrics version

**Comment by nithinraok** [comment]
Currently speaker recognition doesn't work on main due to this bug. While I work on it providing this as a patch.

**Comment by SeanNaren** [comment]
Might be worth making an issue to unpin this in case you forget @nithinraok (unless you're already working on this).
